packages/blade/src/components/Modal/_decisions/decisions.md

```diff
@@ -0,0 +1,227 @@
+- **POC Link**: https://codesandbox.io/s/eloquent-lalande-jvxwno
+  - This POC showcases the responsive behavior of the Modal component.
+  - We've tried 2 approaches with a combination of different `max-width` & `width` values.
+  - Since both of the approaches seem to work as expected, we will implemented the first approach of having `max-width` control the size and a fixed `width` of `100vw - 32px` for all the sizes.
+
+### Modal Height
+- Each Modal will have a max-height of `80vh` (80% of viewport height) and will be scrollable if the content exceeds this height.
+- If the height of the content is less than `80vh`, the Modal's height will be equal to the height of the content.
+
+
+### Modal Usage on mWeb & native apps
+- We will **not be exposing** the Modal component on **mWeb** or native apps (this is a design guideline) instead we would encourage our teams be use BottomSheet for mWeb and native apps.
+- Even though usage on mWeb is discouraged by our design guideline, our Modal will be responsive and will not visually break till 320px on mWeb.
+- We will throw a warning when a Modal is opened on smaller screens as well as add this to our documentation.
+- Designers would need to be mindful of this while designing for mWeb and native apps.
+- On Figma, we have added guardrails to ensure that when screenSize is selected Mobile on Figma, Modal component will turn into a BottomSheet component.
+- We are **not automatically changing** our **Modal to BottomSheet** on code to avoid bundle size overhead as well as to avoid any flow to break unless mindfully implemented with BottomSheet on code.
+
+#### Using BottomSheet on mWeb
+- As per our design guideline, we will be using BottomSheet on mWeb for all the use-cases where we would have used Modal on dWeb.
+- This would require our consumers to add some logic to handle this switch from Modal to BottomSheet on mWeb.
+- This decision also enables our consumers to save on bundle size so that the mWeb's bundle only contains BottomSheet and the dWeb's bundle only contains Modal.
+
+
+**Example usage:**
+
+> You can use Lazy Loading to load the Modal component only on dWeb and BottomSheet component only on mWeb.
+
+```jsx
+import { useTheme, BottomSheet, Modal } from '@razorpay/blade';
+
+const App = () => {
+  const { theme, platform } = useTheme();
+
+  return (
+    <>
+      {platform === 'onMobile' ? (
+        <BottomSheet>
+          <BottomSheetHeader />
+          <BottomSheetBody>
+            /* BottomSheetBody content */
+          </BottomSheetBody>
+          <BottomSheetFooter />
+        </BottomSheet>
+      ) : (
+        <Modal>
+          <ModalHeader />
+          <ModalBody>
+            /* ModalBody content */
+          </ModalBody>
+          <ModalFooter />
+        </Modal>
+      )}
+    </>
+  );
+};
+```
+
+
+### Modal for React Native
+- Since we want to restrict the usage of Modals on native apps, we will not be implementing the Modal component for React Native.
```

Comment:
@chaitanyadeorukhkar For mobile we will be using only bottomsheet going forward?

Reply:
Yeah we will only be using bottomsheet for all use-cases